XbSymbolDatabase.h

```diff
@@ -551,6 +951,73 @@ void XbSymbolContinuousSigScan(bool enable);
 /// <param name="enable">Input boolean to use first symbol address only or not.</param>
 void XbSymbolFirstDetectAddressOnly(bool enable);
 
+// ******************************************************************
+// * API v3 (context) functions to use with other projects.
+// ******************************************************************
+// NOTE: If need to make customize setup, please use the above functions
+// before call XbSymbol_Init function.
+
+/// <summary>
+/// Step 1: Generate library array for LibraryHeader input.
```

Comment:
Ideally the API would imply the order of calls, so we don't have to document it.

Reply:
Well, if use a document tool such as doxygen will generate a documentation for human readability as a help/research tool. Beside, if hover above the API name also provide details too.

Reply:
Well, if use a document tool such as doxygen will generate a documentation for human readability as a help/research tool. Beside, if hover above the API name also provide details too.

Reply:
I still think this implies bad design.

People can also learn the order by looking at the code sample, or it will be implied by what each function is doing (which implies the order).
For example, you can't possibly do: `XbSymbolContext_ScanManual` if you didn't create an object of type `XbSymbolContext` before (using `XbSymbolContext_Create` - which is a misnomer at the moment). This implies an order without having to explicitly document it.

In doxygen, you wouldn't be putting the step information near each function. Instead, you'd keep the generic documentation (order of calls) separated.

Reply:
Right now, there's no sample in this repo. Nor has a tool to for scan xbe files. That's the main reason steps are in the documentations. Later down the road, all of the steps comment will be remove.